ecr/resolver.go

```diff
@@ -184,7 +187,10 @@ func (r *ecrResolver) Resolve(ctx context.Context, ref string) (string, ocispec.
 			WithField("ref", ref).
 			WithField("manifest", manifestBody).
 			Trace("ecr.resolver.resolve: parsing mediaType from manifest")
-		mediaType = parseImageManifestMediaType(ctx, manifestBody)
+		mediaType, err = parseImageManifestMediaType(ctx, manifestBody)
+		if err != nil {
+			return "", ocispec.Descriptor{}, err
+		}
```

Comment:
I can't think of regression cases that might be affected. I'm somewhat concerned that there's one I'm not thinking of right now that @samuelkarp had in mind when we last talked and when the behavior was first added. 

At first blush, it *seems* like we could be preventing alternative mediaType blobs from being transparently passed through (unexpected types, such as non-container image manifests/manifest-lists; eg: [artifacts](https://github.com/opencontainers/artifacts/tree/ecce2ed872459652b173fd5b5f253cdc55abf736)).. but then we only handle that case here if ECR doesn't present the manifest's mediaType in the service's response.

I'll reach out and check in through multiple channels.